components/WeatherForecast.js

```diff
@@ -0,0 +1,118 @@
+import {
+  WeatherForecastField,
+  Overlay,
+  StyledDiv,
+  DisplayedDayDetailsPage,
+  ThreeDaysContainer,
+  WeatherIconDetailsPage,
+  DayContainer,
+  ModalContainer,
+  TemperatureDetailsPage,
+  WeatherHeadline,
+} from "./styledComponents/WeatherForecast.styles";
+
+import { useState, useEffect } from "react";
+import useSWR from "swr";
+import WeekdayFromDateString from "./WeekdayFromDateString";
+import WeatherForecastModal from "./WeatherForecastModal";
+
+const API = process.env.NEXT_PUBLIC_WEATHER_API_KEY;
+
+export default function WeatherForecast({ detailActivity }) {
+  const [filteredWeatherMorning, setFilteredWeatherMorning] = useState([]);
+  const [filteredWeatherAfternoon, setFilteredWeatherAfternoon] = useState([]);
+  const [filteredWeatherEvening, setFilteredWeatherEvening] = useState([]);
+  const [showWeatherForecastModal, setShowWeatherForecastModal] =
+    useState(false);
+
+  const { data: weather } = useSWR(
+    detailActivity.lat &&
+      `https://api.openweathermap.org/data/2.5/forecast?lat=${detailActivity.lat}&lon=${detailActivity.lng}&units=metric&appid=${API}`
+  );
+
+  useEffect(() => {
+    if (weather && weather.list) {
+      const filteredWeather = filterWeatherData(weather.list);
+      setFilteredWeatherMorning(
+        filterWeatherByTime(filteredWeather, "06:00:00")
+      );
+      setFilteredWeatherAfternoon(
+        filterWeatherByTime(filteredWeather, "12:00:00")
+      );
+      setFilteredWeatherEvening(
+        filterWeatherByTime(filteredWeather, "18:00:00")
+      );
+    }
+  }, [weather]);
+
+  // Due to the structure of the weatherObjectArray(the starting point index [0] can vary, for example it can start with a 08:00:00 weather forecast or with a 21:00:00 weather forecast), it is necessary to filter the weatherObjectArray to ensure that the new data includes complete daily weather information. Otherwise, the weather display would lack consistency.
+  function filterWeatherData(weatherList) {
+    const StartingDay = weatherList.findIndex((weatherItem) =>
+      weatherItem.dt_txt.endsWith("06:00:00")
+    );
+    return weatherList.slice(StartingDay);
+  }
+
+  function filterWeatherByTime(weatherList, time) {
+    return weatherList.filter((weatherItem) =>
+      weatherItem.dt_txt.endsWith(time)
+    );
+  }
```

Comment:
very nice functions! Since they don't need any variables from the component scope (fancy term to say that they need to see any variables that are defined inside this component, they only need the input variable weatherlist), you can move them somewhere else and make the code a bit more organised. You can for example move these two functions into a utils.js file or a utils folder with the file filterWeatherData.js